homeassistant/components/zha/core/helpers.py

```diff
@@ -170,10 +171,22 @@ def async_get_zha_device(hass: HomeAssistant, device_id: str) -> ZHADevice:
     device_registry = dr.async_get(hass)
     registry_device = device_registry.async_get(device_id)
     if not registry_device:
+        _LOGGER.error("Device id `%s` not found in registry", device_id)
         raise KeyError(f"Device id `{device_id}` not found in registry.")
     zha_gateway: ZHAGateway = hass.data[DATA_ZHA][DATA_ZHA_GATEWAY]
-    ieee_address = list(list(registry_device.identifiers)[0])[1]
-    ieee = zigpy.types.EUI64.convert(ieee_address)
+    if not zha_gateway.initialized:
+        _LOGGER.error("Attempting to get a ZHA device when ZHA is not initialized")
+        raise RuntimeError("ZHA is not initialized yet")
```

Comment:
Should we raise one of home assistant exceptions instead? IntegrationError or PlatformNotReady error?

Reply:
Probably… switched to use IntegrationError